src/psyclone/tests/psyir/transformations/block_loop_trans_test.py

```diff
@@ -0,0 +1,360 @@
+    assert("Cannot apply a BlockedLoopTrans to a loop where loop variables are"
+           " written to inside the loop body.") in str(excinfo.value)
+
+    # Construct a loop that writes to the variable used for the final value
+    symbol_table = SymbolTable()
+    parent = Loop()
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Literal("512", INTEGER_TYPE))
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    sched = Schedule()
+    parent.addchild(sched)
+    routine = Routine.create("test_routine", symbol_table, [parent])
+    lvar = routine.symbol_table.symbol_from_tag(
+            "lvar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    ivar = routine.symbol_table.symbol_from_tag(
+            "ivar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    binop = BinaryOperation.create(BinaryOperation.Operator.ADD,
+                                   Reference(ivar), Literal("1", INTEGER_TYPE))
+    assign = Assignment.create(Reference(ivar), binop)
+    sched.addchild(assign)
+    parent.children[1].replace_with(Reference(ivar))
+    parent._variable = lvar
+    with pytest.raises(TransformationError) as excinfo:
+        blocktrans.validate(parent)
+    assert("Cannot apply a BlockedLoopTrans to a loop where loop variables are"
+           " written to inside the loop body.") in str(excinfo.value)
+
+
+def test_blockloop_trans_validate5():
+    '''Test the validate function of BlockLoopTrans passes when applying it
+    to a loop that reads from the loop variable'''
+    blocktrans = BlockLoopTrans()
+    # Construct a loop that reads from the loop variable (this is allowed)
+    symbol_table = SymbolTable()
+    parent = Loop()
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Literal("512", INTEGER_TYPE))
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    sched = Schedule()
+    parent.addchild(sched)
+    routine = Routine.create("test_routine", symbol_table, [parent])
+    lvar = routine.symbol_table.symbol_from_tag(
+            "lvar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    ivar = routine.symbol_table.symbol_from_tag(
+            "ivar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    binop = BinaryOperation.create(BinaryOperation.Operator.ADD,
+                                   Reference(lvar), Literal("1", INTEGER_TYPE))
+    assign = Assignment.create(Reference(ivar), binop)
+    sched.addchild(assign)
+    parent._variable = lvar
+    blocktrans.validate(parent)
+
+
+def test_blockloop_trans_apply_pos():
+    '''Test the apply function of BlockLoopTrans'''
```

Comment:
"... for a positive step." And if I have to be picky in all these file docstrings it should say method instead of function :)

Reply:
Done